### ISSUE TITLE
Invalid variable: DC

### ISSUE THREAD

**Issue by jamesragray** [vibe-d/vibe.d#2677]
When trying to use vibe-d 0.9.5 I get the error message "Invalid variable: DC" when compiling with dub build . Using vibe-d 0.9.5-beta.1 works fine.

My dub.selections looks like this:

```json
{
        "fileVersion": 1,
        "versions": {
                "automem": "0.6.7",
                "diet-ng": "1.8.1",
                "eventcore": "0.9.20"
                "libasync": "0.8.6",
                "memutils": "1.0.4",
                "mir-linux-kernel": "1.0.1",
                "openssl": "3.2.2",
                "stdx-allocator": "2.77.5",
                "sumtype": "1.1.4",
                "taggedalgebraic": "0.11.22",
                "test_allocator": "0.3.3",
                "tinyredis": "2.2.1",
                "unit-threaded": "2.0.5",
                "vibe-core": "1.22.4",
                "vibe-d": "0.9.5"
        }
}
```
Here is the version that works:
```json
{
        "fileVersion": 1,
        "versions": {
                "automem": "0.6.7",
                "botan": "1.12.19",
                "botan-math": "1.0.3",
                "diet-ng": "1.8.1",
                "eventcore": "0.9.20",
                "libasync": "0.8.6",
                "memutils": "1.0.4",
                "mir-linux-kernel": "1.0.1",
                "openssl": "1.1.6+1.0.1g",
                "stdx-allocator": "2.77.5",
                "sumtype": "1.1.4",
                "taggedalgebraic": "0.11.22",
                "test_allocator": "0.3.3",
                "tinyredis": "2.2.1",
                "unit-threaded": "2.0.5",
                "vibe-core": "1.22.4",
                "vibe-d": "0.9.5-beta.1"
        }
}
```

**Comment by Andriamanitra** [comment]
Looks like we filed the exact same bug at the same time (#2678) :smile:

**Comment by Geod24** [comment]
This comes from https://github.com/D-Programming-Deimos/openssl/releases/tag/v3.2.1

There are a few possible workarounds:
- Use `library-manual-version` as a `subConfiguration` to `openssl` as described [here](https://github.com/D-Programming-Deimos/openssl#compatibility). Note that you might need to depend explicitly on openssl from your `dub.json` / `dub.sdl` for it to work;
- Fix the openssl version to `<=3.2.0` so that you don't get v3.2.1;
- Disable SSL in Vibe.d (not tested);

I'll look into a proper fix.

**Comment by kinke** [comment]
I was pretty sure dub always defines `DC`: https://github.com/dlang/dub/blob/ead1edac76547bec597b9ea68d3b7c62b7425828/source/dub/generators/generator.d#L1005

**Comment by Geod24** [comment]
@jamesragray : What version of DUB are you using ? @Andriamanitra mentions v1.23.0 which is a bit old (but should still be supported).

**Comment by jamesragray** [comment]
Thanks for the workarounds. My dub is 1.26.1

**Comment by Geod24** [comment]
Can you try with v1.29.0 ? You can get a binary from [here](https://github.com/dlang/dub/releases/tag/v1.29.0).

**Comment by jamesragray** [comment]
My version of gzip doesn't seem to recognize those binaries as gzip files.  So I cloned dub and build v1.29.0. It seems to work with openssl 3.2.2

**Comment by Geod24** [comment]
Confirmed fixed in #2678
